play-json/shared/src/main/scala/Json.scala

```diff
@@ -414,6 +419,29 @@ object Json extends JsonFacade {
   sealed trait MacroOptions
 
   /**
+   * Defines the default macro options.
+   *
+   * Since you can't have defaults for a type parameter (unless it's a contravariant type, it will default to
+   * Nothing), we supply the default via an implicit parameter.
+   */
+  trait DefaultMacroOptions[O <: Json.MacroOptions]
+
+  /**
+   * Low priority implicit used when Json.MacroOptions is not passed.
```

Comment:
The names can be changed. It's an evidence typeclass that specifies what the default type of `Opt <: MacroOptions` should be when `Opt` is not explicitly supplied, working around the fact that the Scala compiler infers `Nothing`. But I'm not sure if there's a better name than `MacroOptionsDefault`, because it is all about the default `MacroOptions` type. We could change `NotDefaultMacroOptions` to `LowPriorityDefaultMacroOptions`, but the thing is that `NotDefaultMacroOptions` actually does supply the `DefaultMacroOptions` type class for `_ <: MacroOptions` types that are not the default, so it seems pretty descriptive to me.